### DISCUSSION TITLE
Error updating Valet

### DISCUSSION THREAD

**Discussion by diogocnunes** [laravel/valet#1454] [Q&A] [answered]
Hi, everyone!

I'm not sure if i caused this issue or what i've to do to solve it, but when i try to update valet using 

```
composer global require laravel/valet 
```

I receive the message below and i don't know what to do to solve it:

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - laravel/valet v4.5.0 conflicts with mnapoli/silly[1.8.1].
    - laravel/valet v4.5.0 conflicts with mnapoli/silly[1.8.1].
    - laravel/valet v4.5.0 conflicts with mnapoli/silly[1.8.1].
    - Installation request for laravel/valet ^4.5 -> satisfiable by laravel/valet[v4.5.0].
    - Installation request for mnapoli/silly (locked at 1.8.1) -> satisfiable by mnapoli/silly[1.8.1].


Installation failed, reverting ./composer.json to its original content.
```

Someone could please help me with it?

Best regards,

Diogo

**Comment by drbyte** [comment]
Can you try this instead?

    composer global update laravel/valet -W

**Reply by diogocnunes** [reply]
With -W option, appears the error below:

```
  [Symfony\Component\Console\Exception\RuntimeException]  
  The "-W" option does not exist.
```

Without the option, no error appears, but nothing was updated:

```
Loading composer repositories with package information
Warning from https://repo.packagist.org: Support for Composer 1 is deprecated and some packages will not be available. You should upgrade to Composer 2. See https://blog.packagist.com/deprecating-composer-1-support/
Updating dependencies (including require-dev)
Nothing to install or update
Package hirak/prestissimo is abandoned, you should avoid using it. No replacement was suggested.
Generating autoload files
19 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
```

But i'm with a older version, since the Valet 4.5.0 was released:

```
~ » valet --version
Laravel Valet 4.3.0
```

I'm using a MacOS Sonoma 14.0 (23A344), if it's important.

**Reply by drbyte** [reply]
> Support for Composer 1 is deprecated

Probably important to get composer updated to v2. I think you can do that with `composer self-update`

> Package hirak/prestissimo is abandoned, you should avoid using it.

Probably wise to run: `composer global remove hirak/prestissimo`

**Reply by drbyte** [reply]
What's in your `~/.composer/composer.json` file?
Are there packages in there that you do not need? Or that you don't remember why they're there? Perhaps make a backup of the file and remove the things you don't need.
(Valet only needs laravel/valet )

**Reply by diogocnunes** [reply]
```
{
    "require": {
        "hirak/prestissimo": "^0.3.10",
        "laravel/valet": "^4.3"
    }
}
```

**Comment by diogocnunes** [comment, marked as answer]
Solved removing the Valet and installing again...
